src/ziggurat/messaging/channel_pool.clj

```diff
@@ -14,9 +14,9 @@
         stream-router-config        (get (zc/ziggurat-config) :stream-router)
         batch-routes-config         (get (zc/ziggurat-config) :batch-routes)
         batch-consumer-thread-count (reduce (fn [sum config]
-                                              (+ sum (:thread-count config))) 0 (vals batch-routes-config))
+                                              (+ sum (get config :thread-count 0))) 0 (vals batch-routes-config))
```

Comment:
I think we should take a default of 1 or a higher number. If no stream-thread-count is specified, this will return a sum of 0?